### PR TITLE
Prep for v1.23.3

### PR DESCRIPTION
**Pull request by odow**
## Pre-release

 - [x] Check that the pinned packages in `docs/Project.toml` are updated. We pin
       the versions so that changes in the solvers (changes in printing, small
       numeric changes) do not break the printing of the JuMP docs in arbitrary
       commits.
 - [x] Check that the `rev` fields in `docs/packages.toml` are updated. We pin
       the versions of solvers and extensions to ensure that changes to their
       READMEs do not break the JuMP docs in arbitrary commits, and to ensure
       that the versions are compatible with the latest JuMP and
       MathOptInterface releases.
 - [x] Check compat of `DimensionalData` in `Project.toml`
 - [x] Check compat of `MacroTools` in `Project.toml`
 - [x] Update `docs/src/changelog.md`
 - [x] https://github.com/jump-dev/JuMP.jl/actions/runs/11431516768
   - [x] PolyJuMP: master not compatible
   - [x] SumOfSquares: master not compatible
   - [x] LinearFractional: https://github.com/nexteraanalytics/LinearFractional.jl/pull/22 
 - [x] Change the version number in `Project.toml`
 - [x] The commit messages in this PR do not contain `[ci skip]`

## The release

 - [ ] After merging this pull request, comment `[at]JuliaRegistrator register` in
       the GitHub commit. This should automatically publish a new version to the
       Julia registry, as well as create a tag, and rebuild the documentation
       for this tag.

       These steps can take quite a bit of time (1 hour or more), so don't be
       surprised if the new documentation takes a while to appear. In addition,
       the links in the README will be broken until JuliaHub fetches the new
       version on their servers.

## Post-release

 - [ ] Once the tag is created, update the relevant `release-` branch. The latest
       release branch at the time of writing is `release-1.0` (we haven't
       back-ported any patches that needed to create a `release-1.Y` branch). To
       to update the release branch with the v1.10.0 tag, do:
       ```
       git checkout release-1.0
       git pull
       git merge v1.10.0
       git push
       ```

### CODE REVIEW

File: docs/Project.toml [hunk 1]
```diff
@@ -45,16 +45,16 @@ CSV = "0.10"
...
-Documenter = "=1.6.0"
+Documenter = "=1.7.0"
 DocumenterCitations = "1"
 Dualization = "0.5"
 Enzyme = "0.13.7"
 ForwardDiff = "0.10"
 GLPK = "=1.2.1"
 HTTP = "1.5.4"
-HiGHS = "=1.9.2"
+HiGHS = "=1.10.0"
```
Comment (odow) [inline docs/Project.toml:57]:
```suggestion
HiGHS = "=1.9.2"
```

Comment (odow) [reply docs/Project.toml:57]: I'll investigate separately.

File: docs/packages.toml [hunk 2]
```diff
@@ -55,9 +55,9 @@
 [GLPK]
     rev = "v1.2.1"
 [Gurobi]
-    rev = "v1.3.0"
+    rev = "v1.3.1"
```
Comment (odow) [inline docs/packages.toml:58]:
```suggestion
    rev = "9bafe7d3e4ed198b97836362a62bb48a14d408bc"
```

File: docs/src/changelog.md [hunk 3]
```diff
@@ -7,6 +7,30 @@ CurrentModule = JuMP
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Version 1.23.3 (October 21, 2024)
+
+### Fixed
+
+ - Fixed a printing bug with scientific nuumbers in `MIME"text/latex"` (#3838)
```
Comment (odow) [inline docs/src/changelog.md:14]:
```suggestion
 - Fixed a printing bug with scientific numbers in `MIME"text/latex"` (#3838)
```

**Review by blegat — APPROVED** [review]

**Pull request merged**